contracts/root/withdrawManager/WithdrawManager.sol

```diff
@@ -47,142 +52,229 @@ contract WithdrawManager is WithdrawManagerStorage, IWithdrawManager {
     bytes calldata withdrawReceipt,
     bytes calldata withdrawReceiptProof
   ) external {
-    address rootToken;
-    uint256 receiptAmountOrTokenId;
-
-    (rootToken, receiptAmountOrTokenId) = _processBurntReceipt(
+    (address rootToken, address childToken, uint256 receiptAmountOrNFTId) = ChildChainVerifier.processBurnReceipt(
       withdrawReceipt,
       path,
       withdrawReceiptProof,
       withdrawBlockReceiptRoot,
       msg.sender
     );
 
-    _processBurntTx(
+    require(
+      registry.rootToChildToken(rootToken) == childToken,
+      "INVALID_ROOT_TO_CHILD_TOKEN_MAPPING"
+    );
+
+    ChildChainVerifier.processBurnTx(
       withdrawTx,
       path,
       withdrawTxProof,
       withdrawBlockTxRoot,
       rootToken,
-      receiptAmountOrTokenId,
-      msg.sender
+      receiptAmountOrNFTId,
+      msg.sender,
+      address(registry), // remove,
+      networkId
     );
 
-    // // exit object
-    // PlasmaExit memory _exitObject = PlasmaExit({
-    //   owner: msg.sender,
-    //   token: rootToken,
-    //   amountOrTokenId: receiptAmountOrTokenId,
-    //   burnt: true
-    // });
+    WithdrawManagerHeader.PlasmaExit memory _exitObject = PlasmaExit({
```

Comment:
no need for `WithdrawManagerHeader.PlasmaExit` instead `PlasmaExit` will work 😛.